subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/CodeNarcActionParameters.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality.internal;
+
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.plugins.quality.CodeNarcReports;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.workers.WorkParameters;
+
+public interface CodeNarcActionParameters extends WorkParameters {
+
+    ConfigurableFileCollection getCompilationClasspath();
+
+    RegularFileProperty getConfig();
+
+    Property<Integer> getMaxPriority1Violations();
+
+    Property<Integer> getMaxPriority2Violations();
+
+    Property<Integer> getMaxPriority3Violations();
+
+    ListProperty<EnabledReport> getEnabledReports();
```

Comment:
Maybe just a `MapProperty<String, RegularFile>`?

Reply:
I like the extra clarity defining an inner domain type provides here, but why is `EnabledReport` not just a simple value class?  What's the need for it to contain `Property`s?

Reply:
@tresat No reason in particular. Perhaps I'm just unsure of exactly how the serialization process works for `WorkParameters`. If we don't need to use `Property`s in this nested class why do we need to use them in the parent `CodeNarcActionParameters`?

Reply:
Same answer as other PR -- using interfaces/managed objects reduces boilerplate. Maybe once we get records we can change this